csrc/ops/arith.cpp

```diff
@@ -895,6 +895,47 @@ NVFUSER_DEFINE_INT_ONLY_OP(bitwise_right_shift, Rshift)
 NVFUSER_DEFINE_INT_ONLY_OP(gcd, Gcd)
 #undef NVFUSER_DEFINE_INT_ONLY_OP
 
+// The logical_right_shift operation shifts the value's bits to the right.
+// If the value is negative, it appends zeros to the front of the value.
+// The sign is preserved with arithmetic_right_shift, so ones are pushed to the
+// most significant bit.
+//
+// An alternate approach is to cast the value to an unsigned integer, perform
+// the right shift, and then cast back to the original value. In C++, unsigned
+// integers are shifted with logical right shift.
+template <typename LHS, typename RHS>
+TORCH_CUDA_CU_API typename std::conditional<
+    std::is_same<LHS, TensorView*>::value ||
+        std::is_same<RHS, TensorView*>::value,
+    TensorView*,
+    Val*>::type
+logical_right_shift_helper(LHS x, RHS shift) {
+  auto number_of_bits = (x->dtype() == PrimDataType::Int) ? 64 : 32;
+  auto one = IrBuilder::create<Int>(x->container(), 1);
+  auto two = IrBuilder::create<Int>(x->container(), 2);
+  auto thirty_two = IrBuilder::create<Int>(x->container(), number_of_bits);
```

Comment:
With the recent refactor https://github.com/NVIDIA/Fuser/pull/529, this now should be:
```suggestion
  auto number_of_bits = (x->dtype() == PrimDataType::Int) ? 64L : 32L;
  auto one = IrBuilder::create<Scalar>(x->container(), 1L);
  auto two = IrBuilder::create<Scalar>(x->container(), 2L);
  auto thirty_two = IrBuilder::create<Scalar>(x->container(), number_of_bits);
```